Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.py

```diff
@@ -0,0 +1,353 @@
+                listformat=list_format,
+                offset=offset,
+            )
+
+            response = client.get_indicators_request(args=query)
+            ipaddr_list.extend(response.get("data", {}).get("ipaddr"))
+            offset += LIMIT
+
+    else:
+        raise DemistoException(f"couldn't fetch, {response.get('success')}, response.get('message')")
+
+    indicators = ipaddr_list
+    demisto.debug(f'{LOG_LINE} got {len(indicators)}')
+
+    results = []
+
+    for indicator in indicators:
+
+        results.append(parse_indicator_for_fetch(indicator, feed_tags, tlp_color, feed_tag_name))
+
+    return now.strftime('%Y-%m-%dT%H:%M:%SZ'), results
+
+
+"""
+MAIN
+"""
+
+
+def main():
+
+    params = demisto.params()
+    """PARAMS"""
+    base_url = urljoin(params["url"], "service/v1/")
+    api_key = params.get("credentials", {}).get("password")
+    verify = not params.get("insecure", False)
+    proxy = params.get("proxy", "")
+    reliability = params.get("integrationReliability", "B - Usually reliable")
+    # TLP 2.0 - CLEAR == WHITE, ++ TLP:AMBER+STRICT
+    tlp_color = params.get('tlp_color') or 'WHITE'
+    feed_tags = params.get('feedTags', '')
+    predefined_edl_name = params['name']
+    list_format = params['list_format'].lower()
+
+    if not DBotScoreReliability.is_valid_type(reliability):
+        raise Exception(
+            "Please provide a valid value for the Source Reliability parameter."
+        )
+
+    try:
+        command = demisto.command()
+        client = Client(
+            base_url=base_url,
+            api_key=api_key,
+            verify=verify,
+            proxy=proxy,
+            reliability=reliability,
+        )
+
+        commands = {
+            "threatvault-get-indicators": threatvault_get_indicators_command
+        }
+        if demisto.command() == "test-module":
+            # This is the call made when pressing the integration Test button.
+            result = client.test_module()
+            return_results(result)
+
+        elif command == 'fetch-indicators':
+            run_datetime, res = fetch_indicators_command(
+                client=client,
+                predefined_edl_name=predefined_edl_name,
+                list_format=list_format,
+                feed_tags=feed_tags,
+                tlp_color=tlp_color,
+            )
+
+            for iter_ in batch(res, batch_size=2000):
+                demisto.debug(f"{LOG_LINE} {iter_=}")
+                demisto.createIndicators(iter_)
+
+            demisto.setLastRun({"last_successful_run": run_datetime})
+
+        elif command in commands:
+            return_results(commands[command](client, list_format, demisto.args()))
+        else:
+            raise NotImplementedError(f'Command "{command}" was not implemented.')
+
+    except Exception:
+        demisto.error(traceback.format_exc())  # print the traceback
+        return_error(f"Failed to execute {command} command. The command not implemented")
```

Comment:
Please note that any exception raised from the code would be excepted here and printed with the "The command not implemented" suffix - which may not always be suitable.

Reply:
True, added better handling of it.